torch/_utils.py

```diff
@@ -393,3 +395,66 @@ def reraise(self):
             # (https://bugs.python.org/issue2651), so we work around it.
             msg = KeyErrorMessage(msg)
         raise self.exc_type(msg)
+
+
+def _get_available_device_type():
+    if torch.cuda.is_available():
+        return "cuda"
+    # add more available device types here
+    return None
+
+
+def _get_device_attr(get_member):
+    device_type = _get_available_device_type()
+    if device_type.lower() == "cuda":
+        return get_member(torch.cuda)
+    # add more available device types here
+    return None
+
+
+def _get_current_device_index():
+    # current device index
+    return _get_device_attr(lambda m: m.current_device())
+
+
+def _get_all_device_indices():
+    # all device index
+    return _get_device_attr(lambda m: list(range(m.device_count())))
+
+
+def _get_devices_properties(device_ids):
+    # all device properties
+    return [_get_device_attr(lambda m: m.get_device_properties(i)) for i in device_ids]
+
+
+def _get_device_index(device, optional=False) -> int:
```

Comment:
You are killing a CUDA device type check which is correct for all current calls of `cuda._utils._get_device_index`. This should be fixed.

Reply:
The type checking has been added for torch.cuda._utils._get_device_index.